### PR TITLE
Second batch of tasks

### PR DESCRIPTION
**Pull request by viktor-viktor**

### CODE REVIEW

File: Viktor_Miroshnychenko/2/run_length_encoding.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    for char in string:
+        if char.isdigit():
+            count = int(char)
+        else:
+            ret_val += char * count
+            count = 1
+
+    return ret_val
+
+
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/run_length_encoding.py]: Too many blank lines

File: Viktor_Miroshnychenko/2/armstrong_numbers.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        digit = int(digit)
+        compare_val += digit**power
+        
+    number = int(number)
+    if number == compare_val:
+        return True
+    else:
+        return False
+
+
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/armstrong_numbers.py]: Too many blank lines

File: Viktor_Miroshnychenko/2/armstrong_numbers.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    :rtype: - bool
+    """
+    if type(number) == int:
+        number = str(number)
+
+    power = len(number)
+    compare_val = 0
+    for digit in number:
+        digit = int(digit)
+        compare_val += digit**power
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/armstrong_numbers.py]: spaces around **

File: Viktor_Miroshnychenko/2/armstrong_numbers.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+def is_amstrong_number(number):
+    """
+    :param number: - number to check weather it is an amstrong number
+    :type number: - str or int
+    
+    :return: - True if param1 is Amstrong number, False otherwise
+    :rtype: - bool
+    """
+    if type(number) == int:
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/armstrong_numbers.py]: Use isinstance( ) instead

File: Viktor_Miroshnychenko/2/armstrong_numbers.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    power = len(number)
+    compare_val = 0
+    for digit in number:
+        digit = int(digit)
+        compare_val += digit**power
+        
+    number = int(number)
+    if number == compare_val:
+        return True
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/armstrong_numbers.py]: Replace with only one return

File: Viktor_Miroshnychenko/2/Uses Filter.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+def my_filter(function, container):
+    return_val = list()
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/Uses Filter.py]: Docstring?

File: Viktor_Miroshnychenko/2/User Map.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+first_arg_list = [1,2,3]
+second_arg_list = [4,5,6,7]
+
+def my_map(function, *containers):
+    minLen = len(containers[0])
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/User Map.py]: Docstring?

File: Viktor_Miroshnychenko/2/User Map.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    for arg in containers:
+        if minLen > len(arg):
+            minLen = len(arg)
+
+    func_arg_list = list()
+    return_val = list()
+    for i in range(minLen):
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/User Map.py]: naming_convention?

File: Viktor_Miroshnychenko/2/Mapping a list of functions.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+def map_functions(variable, functions):
+    return_val = list()
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/Mapping a list of functions.py]: It's better to use [ ]

File: Viktor_Miroshnychenko/2/Allergies.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
+def find_allergis(total_score):
+    aler_score = [1,2,4,8,16,32,64,128]
+    aler_names = ["eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"]
+    aler_score.reverse()
+    aler_names.reverse()
+
+    return_val = list()
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/Allergies.py]: It's better to use [ ]

File: Viktor_Miroshnychenko/2/Allergies.py [hunk 11]
```diff
@@ -0,0 +1,17 @@
+def find_allergis(total_score):
+    aler_score = [1,2,4,8,16,32,64,128]
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/Allergies.py]: spaces after commas

Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/Allergies.py]: Docstring?

File: Viktor_Miroshnychenko/2/Mapping a list of functions.py (hunk 9 above)
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/2/Mapping a list of functions.py]: Docstring?

File: Viktor_Miroshnychenko/3/crypto_square.py [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    while height - width > 1:
+        width += 1
+        height = number // width
+        if number % width > 0:
+            height += 1
+
+    if height > width:
+        height, width = width, height
+
+
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/3/crypto_square.py]: One blank line here.

File: Viktor_Miroshnychenko/3/crypto_square.py [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    return [width, height]
+    
+
+def encode(string):
+    """
+    :param string: string to encode
+    :type string: str
+
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/3/crypto_square.py]: No blank line

File: Viktor_Miroshnychenko/3/crypto_square.py [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+    for count in range(rect[0]):
+        i = count
+        while i < len(string):
+            encoded_string += string[i]
+            i += rect[0]
+
+    print(encoded_string)
+    return encoded_string
+        
+
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/3/crypto_square.py]: 2 blank lines

File: Viktor_Miroshnychenko/3/poker.py [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+    :type hands: list[str]
+
+    :return: highest hand
+    :rtype: str
+    """
+
+    power_list = []
+    for index in range(len(hands)):
+         power_list.append(0)
+    for index in range(len(hands)):
```
Comment (oleksandr-medviediev) [inline Viktor_Miroshnychenko/3/poker.py]: There's much better way to do this.

Comment (viktor-viktor) [reply Viktor_Miroshnychenko/3/poker.py]: do you mean the whole task or that particular function ?

**Pull request merged**